[Downloader] Cannot open downloaded file - SecurityException: does not have permission to content

### Checklist

- [x] I am able to reproduce the bug with the latest version given here: [CLICK THIS LINK](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [x] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [x] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [x] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [x] This issue contains only one bug.
- [x] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).
- [x] I have read and understood the [AI policy](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md#ai-policy). The content of this bug report is not generated by AI.

### Affected version

## Exception * __User Action:__ ui error * __Request:__ ACRA report * __Content Country:__ US * __Content Language:__ en-US * __App Language:__ en_US * __Service:__ <unknown_service> * __Timestamp:__ 2026-01-26T11:36:16.265+05:30 * __Package:__ org.schabi.newpipe * __Service:__ <unknown_service> * __Version:__ 0.28.1 * __OS:__ Linux Android 14 - 34 <details><summary><b>Crash log </b></summary><p>  ``` java.lang.SecurityException: UID 10871 does not have permission to content://com.android.externalstorage.documents/document/primary%3ADownload%2FComplete%20Class%2012th%20Maths%20%F0%9F%93%96%F0%9F%94%A5%20_%20ALL%20Chapters%20in%20One%20Video%20!.mp4 [user 0] [prefix]; you could obtain access using ACTION_OPEN_DOCUMENT or related APIs 	at android.os.Parcel.createExceptionOrNull(Parcel.java:3089) 	at android.os.Parcel.createException(Parcel.java:3073) 	at android.os.Parcel.readException(Parcel.java:3056) 	at android.os.Parcel.readException(Parcel.java:2998) 	at android.app.IActivityTaskManager$Stub$Proxy.startActivity(IActivityTaskManager.java:3802) 	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1880) 	at android.app.Activity.startActivityForResult(Activity.java:5626) 	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:780) 	at android.app.Activity.startActivityForResult(Activity.java:5584) 	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:761) 	at android.app.Activity.startActivity(Activity.java:6082) 	at android.app.Activity.startActivity(Activity.java:6049) 	at org.schabi.newpipe.util.external_communication.ShareUtils.tryOpenIntentInApp(ShareUtils.java:128) 	at org.schabi.newpipe.util.external_communication.ShareUtils.openIntentInApp(ShareUtils.java:146) 	at us.shandian.giga.ui.adapter.MissionAdapter.viewWithFileProvider(MissionAdapter.java:361) 	at us.shandian.giga.ui.adapter.MissionAdapter.-$$Nest$mviewWithFileProvider(MissionAdapter.java:0) 	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.lambda$new$1(MissionAdapter.java:910) 	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.$r8$lambda$lUyMuzEHgx7VO8u56WYBu3QggdM(MissionAdapter.java:0) 	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem$$ExternalSyntheticLambda1.onClick(R8$$SyntheticClass:0) 	at android.view.View.performClick(View.java:7677) 	at android.view.View.performClickInternal(View.java:7650) 	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0) 	at android.view.View$PerformClick.run(View.java:30412) 	at android.os.Handler.handleCallback(Handler.java:958) 	at android.os.Handler.dispatchMessage(Handler.java:99) 	at android.os.Looper.loopOnce(Looper.java:243) 	at android.os.Looper.loop(Looper.java:338) 	at android.app.ActivityThread.main(ActivityThread.java:8470) 	at java.lang.reflect.Method.invoke(Native Method) 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600) 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1064) Caused by: android.os.RemoteException: Remote stack trace: 	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionUnlocked(UriGrantsManagerService.java:1263) 	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:642) 	at com.android.server.uri.UriGrantsManagerService.-$$Nest$mcheckGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:0) 	at com.android.server.uri.UriGrantsManagerService$LocalService.checkGrantUriPermissionFromIntent(UriGrantsManagerService.java:1493) 	at com.android.server.wm.ActivityStarter$Request.resolveActivity(ActivityStarter.java:645)   ``` </details> <hr>

### Steps to reproduce the bug

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ US
* __Content Language:__ en-US
* __App Language:__ en_US
* __Service:__ <unknown_service>
* __Timestamp:__ 2026-01-26T11:36:16.265+05:30
* __Package:__ org.schabi.newpipe
* __Service:__ <unknown_service>
* __Version:__ 0.28.1
* __OS:__ Linux Android 14 - 34
<details><summary><b>Crash log </b></summary><p>

```
java.lang.SecurityException: UID 10871 does not have permission to content://com.android.externalstorage.documents/document/primary%3ADownload%2FComplete%20Class%2012th%20Maths%20%F0%9F%93%96%F0%9F%94%A5%20_%20ALL%20Chapters%20in%20One%20Video%20!.mp4 [user 0] [prefix]; you could obtain access using ACTION_OPEN_DOCUMENT or related APIs
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3089)
	at android.os.Parcel.createException(Parcel.java:3073)
	at android.os.Parcel.readException(Parcel.java:3056)
	at android.os.Parcel.readException(Parcel.java:2998)
	at android.app.IActivityTaskManager$Stub$Proxy.startActivity(IActivityTaskManager.java:3802)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1880)
	at android.app.Activity.startActivityForResult(Activity.java:5626)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:780)
	at android.app.Activity.startActivityForResult(Activity.java:5584)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:761)
	at android.app.Activity.startActivity(Activity.java:6082)
	at android.app.Activity.startActivity(Activity.java:6049)
	at org.schabi.newpipe.util.external_communication.ShareUtils.tryOpenIntentInApp(ShareUtils.java:128)
	at org.schabi.newpipe.util.external_communication.ShareUtils.openIntentInApp(ShareUtils.java:146)
	at us.shandian.giga.ui.adapter.MissionAdapter.viewWithFileProvider(MissionAdapter.java:361)
	at us.shandian.giga.ui.adapter.MissionAdapter.-$$Nest$mviewWithFileProvider(MissionAdapter.java:0)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.lambda$new$1(MissionAdapter.java:910)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.$r8$lambda$lUyMuzEHgx7VO8u56WYBu3QggdM(MissionAdapter.java:0)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem$$ExternalSyntheticLambda1.onClick(R8$$SyntheticClass:0)
	at android.view.View.performClick(View.java:7677)
	at android.view.View.performClickInternal(View.java:7650)
	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
	at android.view.View$PerformClick.run(View.java:30412)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:243)
	at android.os.Looper.loop(Looper.java:338)
	at android.app.ActivityThread.main(ActivityThread.java:8470)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1064)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionUnlocked(UriGrantsManagerService.java:1263)
	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:642)
	at com.android.server.uri.UriGrantsManagerService.-$$Nest$mcheckGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:0)
	at com.android.server.uri.UriGrantsManagerService$LocalService.checkGrantUriPermissionFromIntent(UriGrantsManagerService.java:1493)
	at com.android.server.wm.ActivityStarter$Request.resolveActivity(ActivityStarter.java:645)


```
</details>
<hr>


### Expected behavior

_No response_

### Actual behavior

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ US
* __Content Language:__ en-US
* __App Language:__ en_US
* __Service:__ <unknown_service>
* __Timestamp:__ 2026-01-26T11:36:16.265+05:30
* __Package:__ org.schabi.newpipe
* __Service:__ <unknown_service>
* __Version:__ 0.28.1
* __OS:__ Linux Android 14 - 34
<details><summary><b>Crash log </b></summary><p>

```
java.lang.SecurityException: UID 10871 does not have permission to content://com.android.externalstorage.documents/document/primary%3ADownload%2FComplete%20Class%2012th%20Maths%20%F0%9F%93%96%F0%9F%94%A5%20_%20ALL%20Chapters%20in%20One%20Video%20!.mp4 [user 0] [prefix]; you could obtain access using ACTION_OPEN_DOCUMENT or related APIs
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3089)
	at android.os.Parcel.createException(Parcel.java:3073)
	at android.os.Parcel.readException(Parcel.java:3056)
	at android.os.Parcel.readException(Parcel.java:2998)
	at android.app.IActivityTaskManager$Stub$Proxy.startActivity(IActivityTaskManager.java:3802)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1880)
	at android.app.Activity.startActivityForResult(Activity.java:5626)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:780)
	at android.app.Activity.startActivityForResult(Activity.java:5584)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:761)
	at android.app.Activity.startActivity(Activity.java:6082)
	at android.app.Activity.startActivity(Activity.java:6049)
	at org.schabi.newpipe.util.external_communication.ShareUtils.tryOpenIntentInApp(ShareUtils.java:128)
	at org.schabi.newpipe.util.external_communication.ShareUtils.openIntentInApp(ShareUtils.java:146)
	at us.shandian.giga.ui.adapter.MissionAdapter.viewWithFileProvider(MissionAdapter.java:361)
	at us.shandian.giga.ui.adapter.MissionAdapter.-$$Nest$mviewWithFileProvider(MissionAdapter.java:0)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.lambda$new$1(MissionAdapter.java:910)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.$r8$lambda$lUyMuzEHgx7VO8u56WYBu3QggdM(MissionAdapter.java:0)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem$$ExternalSyntheticLambda1.onClick(R8$$SyntheticClass:0)
	at android.view.View.performClick(View.java:7677)
	at android.view.View.performClickInternal(View.java:7650)
	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
	at android.view.View$PerformClick.run(View.java:30412)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:243)
	at android.os.Looper.loop(Looper.java:338)
	at android.app.ActivityThread.main(ActivityThread.java:8470)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1064)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionUnlocked(UriGrantsManagerService.java:1263)
	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:642)
	at com.android.server.uri.UriGrantsManagerService.-$$Nest$mcheckGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:0)
	at com.android.server.uri.UriGrantsManagerService$LocalService.checkGrantUriPermissionFromIntent(UriGrantsManagerService.java:1493)
	at com.android.server.wm.ActivityStarter$Request.resolveActivity(ActivityStarter.java:645)


```
</details>
<hr>


### Screenshots/Screen recordings

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ US
* __Content Language:__ en-US
* __App Language:__ en_US
* __Service:__ <unknown_service>
* __Timestamp:__ 2026-01-26T11:36:16.265+05:30
* __Package:__ org.schabi.newpipe
* __Service:__ <unknown_service>
* __Version:__ 0.28.1
* __OS:__ Linux Android 14 - 34
<details><summary><b>Crash log </b></summary><p>

```
java.lang.SecurityException: UID 10871 does not have permission to content://com.android.externalstorage.documents/document/primary%3ADownload%2FComplete%20Class%2012th%20Maths%20%F0%9F%93%96%F0%9F%94%A5%20_%20ALL%20Chapters%20in%20One%20Video%20!.mp4 [user 0] [prefix]; you could obtain access using ACTION_OPEN_DOCUMENT or related APIs
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3089)
	at android.os.Parcel.createException(Parcel.java:3073)
	at android.os.Parcel.readException(Parcel.java:3056)
	at android.os.Parcel.readException(Parcel.java:2998)
	at android.app.IActivityTaskManager$Stub$Proxy.startActivity(IActivityTaskManager.java:3802)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1880)
	at android.app.Activity.startActivityForResult(Activity.java:5626)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:780)
	at android.app.Activity.startActivityForResult(Activity.java:5584)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:761)
	at android.app.Activity.startActivity(Activity.java:6082)
	at android.app.Activity.startActivity(Activity.java:6049)
	at org.schabi.newpipe.util.external_communication.ShareUtils.tryOpenIntentInApp(ShareUtils.java:128)
	at org.schabi.newpipe.util.external_communication.ShareUtils.openIntentInApp(ShareUtils.java:146)
	at us.shandian.giga.ui.adapter.MissionAdapter.viewWithFileProvider(MissionAdapter.java:361)
	at us.shandian.giga.ui.adapter.MissionAdapter.-$$Nest$mviewWithFileProvider(MissionAdapter.java:0)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.lambda$new$1(MissionAdapter.java:910)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.$r8$lambda$lUyMuzEHgx7VO8u56WYBu3QggdM(MissionAdapter.java:0)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem$$ExternalSyntheticLambda1.onClick(R8$$SyntheticClass:0)
	at android.view.View.performClick(View.java:7677)
	at android.view.View.performClickInternal(View.java:7650)
	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
	at android.view.View$PerformClick.run(View.java:30412)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:243)
	at android.os.Looper.loop(Looper.java:338)
	at android.app.ActivityThread.main(ActivityThread.java:8470)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1064)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionUnlocked(UriGrantsManagerService.java:1263)
	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:642)
	at com.android.server.uri.UriGrantsManagerService.-$$Nest$mcheckGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:0)
	at com.android.server.uri.UriGrantsManagerService$LocalService.checkGrantUriPermissionFromIntent(UriGrantsManagerService.java:1493)
	at com.android.server.wm.ActivityStarter$Request.resolveActivity(ActivityStarter.java:645)


```
</details>
<hr>


### Logs

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ US
* __Content Language:__ en-US
* __App Language:__ en_US
* __Service:__ <unknown_service>
* __Timestamp:__ 2026-01-26T11:36:16.265+05:30
* __Package:__ org.schabi.newpipe
* __Service:__ <unknown_service>
* __Version:__ 0.28.1
* __OS:__ Linux Android 14 - 34
<details><summary><b>Crash log </b></summary><p>

```
java.lang.SecurityException: UID 10871 does not have permission to content://com.android.externalstorage.documents/document/primary%3ADownload%2FComplete%20Class%2012th%20Maths%20%F0%9F%93%96%F0%9F%94%A5%20_%20ALL%20Chapters%20in%20One%20Video%20!.mp4 [user 0] [prefix]; you could obtain access using ACTION_OPEN_DOCUMENT or related APIs
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3089)
	at android.os.Parcel.createException(Parcel.java:3073)
	at android.os.Parcel.readException(Parcel.java:3056)
	at android.os.Parcel.readException(Parcel.java:2998)
	at android.app.IActivityTaskManager$Stub$Proxy.startActivity(IActivityTaskManager.java:3802)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1880)
	at android.app.Activity.startActivityForResult(Activity.java:5626)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:780)
	at android.app.Activity.startActivityForResult(Activity.java:5584)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:761)
	at android.app.Activity.startActivity(Activity.java:6082)
	at android.app.Activity.startActivity(Activity.java:6049)
	at org.schabi.newpipe.util.external_communication.ShareUtils.tryOpenIntentInApp(ShareUtils.java:128)
	at org.schabi.newpipe.util.external_communication.ShareUtils.openIntentInApp(ShareUtils.java:146)
	at us.shandian.giga.ui.adapter.MissionAdapter.viewWithFileProvider(MissionAdapter.java:361)
	at us.shandian.giga.ui.adapter.MissionAdapter.-$$Nest$mviewWithFileProvider(MissionAdapter.java:0)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.lambda$new$1(MissionAdapter.java:910)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.$r8$lambda$lUyMuzEHgx7VO8u56WYBu3QggdM(MissionAdapter.java:0)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem$$ExternalSyntheticLambda1.onClick(R8$$SyntheticClass:0)
	at android.view.View.performClick(View.java:7677)
	at android.view.View.performClickInternal(View.java:7650)
	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
	at android.view.View$PerformClick.run(View.java:30412)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:243)
	at android.os.Looper.loop(Looper.java:338)
	at android.app.ActivityThread.main(ActivityThread.java:8470)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1064)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionUnlocked(UriGrantsManagerService.java:1263)
	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:642)
	at com.android.server.uri.UriGrantsManagerService.-$$Nest$mcheckGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:0)
	at com.android.server.uri.UriGrantsManagerService$LocalService.checkGrantUriPermissionFromIntent(UriGrantsManagerService.java:1493)
	at com.android.server.wm.ActivityStarter$Request.resolveActivity(ActivityStarter.java:645)


```
</details>
<hr>

- [ ] 

- [ ] 

### Affected Android/Custom ROM version

_No response_

### Affected device model

_No response_

### Additional information

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ US
* __Content Language:__ en-US
* __App Language:__ en_US
* __Service:__ <unknown_service>
* __Timestamp:__ 2026-01-26T11:36:16.265+05:30
* __Package:__ org.schabi.newpipe
* __Service:__ <unknown_service>
* __Version:__ 0.28.1
* __OS:__ Linux Android 14 - 34
<details><summary><b>Crash log </b></summary><p>

```
java.lang.SecurityException: UID 10871 does not have permission to content://com.android.externalstorage.documents/document/primary%3ADownload%2FComplete%20Class%2012th%20Maths%20%F0%9F%93%96%F0%9F%94%A5%20_%20ALL%20Chapters%20in%20One%20Video%20!.mp4 [user 0] [prefix]; you could obtain access using ACTION_OPEN_DOCUMENT or related APIs
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3089)
	at android.os.Parcel.createException(Parcel.java:3073)
	at android.os.Parcel.readException(Parcel.java:3056)
	at android.os.Parcel.readException(Parcel.java:2998)
	at android.app.IActivityTaskManager$Stub$Proxy.startActivity(IActivityTaskManager.java:3802)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1880)
	at android.app.Activity.startActivityForResult(Activity.java:5626)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:780)
	at android.app.Activity.startActivityForResult(Activity.java:5584)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:761)
	at android.app.Activity.startActivity(Activity.java:6082)
	at android.app.Activity.startActivity(Activity.java:6049)
	at org.schabi.newpipe.util.external_communication.ShareUtils.tryOpenIntentInApp(ShareUtils.java:128)
	at org.schabi.newpipe.util.external_communication.ShareUtils.openIntentInApp(ShareUtils.java:146)
	at us.shandian.giga.ui.adapter.MissionAdapter.viewWithFileProvider(MissionAdapter.java:361)
	at us.shandian.giga.ui.adapter.MissionAdapter.-$$Nest$mviewWithFileProvider(MissionAdapter.java:0)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.lambda$new$1(MissionAdapter.java:910)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem.$r8$lambda$lUyMuzEHgx7VO8u56WYBu3QggdM(MissionAdapter.java:0)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem$$ExternalSyntheticLambda1.onClick(R8$$SyntheticClass:0)
	at android.view.View.performClick(View.java:7677)
	at android.view.View.performClickInternal(View.java:7650)
	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
	at android.view.View$PerformClick.run(View.java:30412)
	at android.os.Handler.handleCallback(Handler.java:958)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:243)
	at android.os.Looper.loop(Looper.java:338)
	at android.app.ActivityThread.main(ActivityThread.java:8470)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:600)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1064)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionUnlocked(UriGrantsManagerService.java:1263)
	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:642)
	at com.android.server.uri.UriGrantsManagerService.-$$Nest$mcheckGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:0)
	at com.android.server.uri.UriGrantsManagerService$LocalService.checkGrantUriPermissionFromIntent(UriGrantsManagerService.java:1493)
	at com.android.server.wm.ActivityStarter$Request.resolveActivity(ActivityStarter.java:645)


```
</details>
<hr>

Here is a crash report which contains most of the requested details.

## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ US
* __Content Language:__ en-US
* __App Language:__ en_US_#u-mu-celsius
* __Service:__ <unknown_service>
* __Timestamp:__ 2026-03-29T21:23:28.516+02:00
* __Package:__ org.schabi.newpipe
* __Service:__ <unknown_service>
* __Version:__ 0.28.4
* __OS:__ Linux Android 16 - 36
<details><summary><b>Crash log </b></summary><p>

```
java.lang.SecurityException: UID 10325 does not have permission to content://com.android.providers.downloads.documents/document/raw%3A%2Fstorage%2Femulated%2F0%2FDownload%2FNewPipe%2Fpodcasts%2FBezpe%C4%8Dnostn%C3%BD%20radar_%20Slovensko%20si%20mus%C3%AD%20za%C4%8Da%C5%A5%20ve%C4%BEmi%20r%C3%BDchlo%20vyhodnocova%C5%A5%20svoje%20strategick%C3%A9%20z%C3%A1ujmy.m4a [user 0] [prefix]; you could obtain access using ACTION_OPEN_DOCUMENT or related APIs
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3370)
	at android.os.Parcel.createException(Parcel.java:3354)
	at android.os.Parcel.readException(Parcel.java:3337)
	at android.os.Parcel.readException(Parcel.java:3279)
	at android.app.IActivityTaskManager$Stub$Proxy.startActivity(IActivityTaskManager.java:2382)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:2017)
	at android.app.Activity.startActivityForResult(Activity.java:6059)
	at android.app.Activity.startActivityForResult(Activity.java:6016)
	at android.app.Activity.startActivity(Activity.java:6519)
	at android.app.Activity.startActivity(Activity.java:6486)
	at us.shandian.giga.ui.adapter.MissionAdapter$ViewHolderItem$$ExternalSyntheticLambda0.onClick(r8-map-id-ce0307bac2c3f16e8dc2f832773b0599f307ee51c82b5550b73911d8a6d3ce37:68)
	at android.view.View.performClick(View.java:8229)
	at android.view.View.performClickInternal(View.java:8206)
	at android.view.View.-$$Nest$mperformClickInternal(Unknown Source:0)
	at android.view.View$PerformClick.run(View.java:32007)
	at android.os.Handler.handleCallback(Handler.java:1095)
	at android.os.Handler.dispatchMessageImpl(Handler.java:135)
	at android.os.Handler.dispatchMessage(Handler.java:125)
	at android.os.Looper.loopOnce(Looper.java:269)
	at android.os.Looper.loop(Looper.java:367)
	at android.app.ActivityThread.main(ActivityThread.java:9333)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:566)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:929)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.uri.UriGrantsManagerService$$ExternalSyntheticBUOutline0.m(R8$$SyntheticClass:0)
	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionUnlocked(UriGrantsManagerService.java:1448)
	at com.android.server.uri.UriGrantsManagerService.checkGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:709)
	at com.android.server.uri.UriGrantsManagerService.-$$Nest$mcheckGrantUriPermissionFromIntentUnlocked(UriGrantsManagerService.java:0)
	at com.android.server.uri.UriGrantsManagerService$LocalService.internalCheckGrantUriPermissionFromIntent(UriGrantsManagerService.java:1769)


```
</details>
<hr>

The downloaded files are stored in subfolders of the main Downloads directory. I would like to open then in VLC. But I never get a chance to even pick an application to open them in because NewPipe crashes before that. Although it is possible that the default application to open the video and audio files has already been set.

VLC has full photos and videos permissions but I have also tired it without allowing VLC this permission. The outcome was the same.

I have also tried the version of the application that you have attached - thank you for that. That app does not crash. But it also would not open the downloaded file. What happens instead is that it would open the file browser at the default location.

--- a/app/src/main/java/us/shandian/giga/ui/adapter/MissionAdapter.java
+++ b/app/src/main/java/us/shandian/giga/ui/adapter/MissionAdapter.java
@@ -1,5 +1,6 @@
 package us.shandian.giga.ui.adapter;
 
+import static android.content.Intent.FLAG_ACTIVITY_NEW_TASK;
 import static android.content.Intent.FLAG_GRANT_PREFIX_URI_PERMISSION;
 import static android.content.Intent.FLAG_GRANT_READ_URI_PERMISSION;
 import static android.content.Intent.createChooser;
@@ -356,7 +357,9 @@ public class MissionAdapter extends Adapter<ViewHolder> implements Handler.Callb
         viewIntent.addFlags(FLAG_GRANT_PREFIX_URI_PERMISSION);
 
         Intent chooserIntent = createChooser(viewIntent, null);
-        chooserIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | FLAG_GRANT_READ_URI_PERMISSION);
+        chooserIntent.addFlags(FLAG_ACTIVITY_NEW_TASK);
+        chooserIntent.addFlags(FLAG_GRANT_READ_URI_PERMISSION);
+        chooserIntent.addFlags(FLAG_GRANT_PREFIX_URI_PERMISSION);
 
         ShareUtils.openIntentInApp(mContext, chooserIntent);
     }
@@ -375,7 +378,7 @@ public class MissionAdapter extends Adapter<ViewHolder> implements Handler.Callb
         if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.O_MR1) {
             intent.putExtra(Intent.EXTRA_TITLE, mContext.getString(R.string.share_dialog_title));
         }
-        intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        intent.addFlags(FLAG_ACTIVITY_NEW_TASK);
         intent.addFlags(FLAG_GRANT_READ_URI_PERMISSION);
 
         mContext.startActivity(intent);

